pkgs/servers/mail/dovecot/default.nix

```diff
@@ -44,6 +47,11 @@ stdenv.mkDerivation rec {
     # so we can symlink plugins from several packages there.
     # The symlinking needs to be done in NixOS.
     ./2.3.x-module_dir.patch
+    # This fixes the build of the lua module, remove when it's included in a relase
+    (fetchpatch {
+      url = "https://github.com/dovecot/core/commit/2dac24558091c673e5918568d9958f6d8f261fc5.patch";
+      sha256 = "05hglw36imq4858055a48j0gnwm85g8z6kdckjghsm69bc97y5n2";
+    })
   ];
```

Comment:
```suggestion
  ] ++ lib.optionals withLua [
    # This fixes the build of the lua module, remove when it's included in a relase
    (fetchpatch {
      url = "https://github.com/dovecot/core/commit/2dac24558091c673e5918568d9958f6d8f261fc5.patch";
      sha256 = "05hglw36imq4858055a48j0gnwm85g8z6kdckjghsm69bc97y5n2";
    })
  ];
```

Reply:
This is a one-line patch to `src/lib-lua/Makefile.am`. Making it optional really doesn't make much sense.

Reply:
Appling patches optional makes them bitrot very easily. Generally this should be avoided as much as possible.